modules/order/src/Adjustment.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Drupal\commerce_order;
+use Drupal\commerce_price\Price;
+
+/**
+ * Represents an adjustment.
+ */
+final class Adjustment {
+
+  /**
+   * The adjustment type.
+   *
+   * @var string
+   */
+  protected $type;
+
+  /**
+   * The adjustment label.
+   *
+   * @var string
+   */
+  protected $label;
+
+  /**
+   * The decimal amount.
+   *
+   * @var string
+   */
+  protected $amount;
+
+  /**
+   * The currency code.
+   *
+   * @var string
+   */
+  protected $currencyCode;
+
+  /**
+   * The source identifier of the adjustment.
+   *
+   * @var string
+   */
+  protected $sourceId;
+
+  /**
+   * Constructs a new Adjustment instance.
+   *
+   * @param array $definition
+   *   The definition.
+   */
+  public function __construct(array $definition) {
+    foreach (['type', 'label', 'amount', 'currency_code', 'source_id'] as $required_property) {
+      if (empty($definition[$required_property])) {
+        throw new \InvalidArgumentException(sprintf('Missing required property %s.', $required_property));
+      }
+    }
+
+    $this->type = $definition['type'];
+    $this->label = $definition['label'];
+    $this->amount = $definition['amount'];
+    $this->currencyCode = $definition['currency_code'];
+    $this->sourceId = $definition['source_id'];
+  }
+
+  /**
+   * Gets the adjustment type.
+   *
+   * @return string
+   *   The adjustment type.
+   */
+  public function getType() {
+    return $this->type;
+  }
+
+  /**
+   * Gets the adjustment label.
+   *
+   * @return string
+   *   The adjustment label.
+   */
+  public function getLabel() {
+    return $this->label;
+  }
+
+  /**
+   * Gets the decimal amount.
+   *
+   * @return string
+   *   The decimal amount.
+   */
+  public function getDecimalAmount() {
```

Comment:
Then maybe we rename the property to $decimalAmount for clarity's sake?

> On Jul 23, 2016, at 6:14 AM, Bojan Zivanovic <notifications@github.com> wrote:
> 
> In modules/order/src/Adjustment.php:
> 
> > +   * Gets the adjustment label.
> > +   *
> > +   * @return string
> > +   *   The adjustment label.
> > +   */
> > +  public function getLabel() {
> > +    return $this->label;
> > +  }
> > +
> > +  /**
> > +   * Gets the decimal amount.
> > +   *
> > +   * @return string
> > +   *   The decimal amount.
> > +   */
> > +  public function getDecimalAmount() {
> It's getDecimalAmount() on Price, Matt copied it to Adjustment for consistency. There we went with getDecimalAmount() to differentiate between amount the number and amount the object (e.g no $amount->getAmount()). So we probably need to leave this for now.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>